### PR TITLE
Two improvements

### PR DESCRIPTION
**Pull request by DaanDeMeyer**

### CODE REVIEW

File: mkosi/run.py [hunk 1]
```diff
@@ -435,11 +435,23 @@ def vartmpdir() -> Iterator[Path]:
...
+                def remove() -> None:
+                    acquire_privileges()
+                    shutil.rmtree(d)
+
+                fork_and_wait(remove)
+                return
+            elif e.errno != errno.ENOENT:
+                raise
 
         shutil.rmtree(d)
```
Comment (behrmann) [inline mkosi/run.py:456]: I think by using the `else` on the try except the warning about the return in the except could be shut up as well. 
```suggestion
        except OSError as e:
            if e.errno == errno.ENOTEMPTY:
                def remove() -> None:
                    acquire_privileges()
                    shutil.rmtree(d)

                fork_and_wait(remove)
            elif e.errno != errno.ENOENT:
                raise
        else:
            shutil.rmtree(d)
```

**Review by behrmann — APPROVED** [review]
LGTM, one alternative for the warning. It' a bit of the trade off and I don't think it would be necessarily better, because it could be misread as being on the if-elif-block.

File: mkosi/run.py (hunk 1 above)
Comment (DaanDeMeyer) [reply mkosi/run.py:456]: Doesn't work, the else block is only executed if the try block finishes without raising an exception

Comment (behrmann) [reply mkosi/run.py:456]: You're right, I wrongly assumed we could only fall through if there are not exceptions, but `ENOENT` is silently skipped. This shouldn't be bad, but it's a bit unexpected, so maybe this should be logged.

**Pull request merged**